src/app/shared/components/course-form/course-form.component.html

```diff
@@ -19,38 +24,97 @@ <h3>Main Info</h3>
           placeholder="Input text"
           rows="4"
           cols="5"
+          formControlName="description"
         ></textarea>
-        <span id="descriptionErrorMessage" class="text-danger">Description is required.</span>
-      </div>
-      <div class="separator"></div>
-      <h3>Authors</h3>
-      <div class="form__control">
-        <div class="course__flex">
-          <!--Use the name `author` for the form control.-->
-          <input type="text" id="author" placeholder="Input Author Name"/>
-          <app-button id="createAuthor"></app-button>
-        </div>
-        <span id="authorErrorMessage">New author should contain only latin letters and numbers.</span>
-        <div class="course__authors">
-          <!--Use the name `authors` for the form array.-->
-          ::Authors::
-        </div>
+        <ng-container *ngIf="(submitted || description?.touched) && description?.hasError('required')">
+          <span id="descriptionErrorMessage" class="text-danger">Description is required.</span>
+        </ng-container>
+        <ng-container *ngIf="(submitted || description?.touched) && description?.hasError('minlength')">
+          <span id="descriptionErrorMessage" class="text-danger">Description must be minimum 2 characters long.</span>
+        </ng-container>
       </div>
       <div class="separator"></div>
       <h3>Duration</h3>
       <div class="form__control">
         <div class="course__flex">
-          <!--Use the name `duration` for the form array.-->
-          <input type="number" id="duration" placeholder="Input duration"/>
+          <input type="number" id="duration" placeholder="Minutes..." formControlName="duration"/>
           <div class="course__duration">
-            ::Duration::
+            {{ duration?.value ? (duration?.value | duration) : (0 | duration) }}
+          </div>
+        </div>
+        <ng-container *ngIf="(submitted || duration?.touched) && duration?.hasError('required')">
+          <span id="durationErrorMessage" class="text-danger">Duration is required.</span>
+        </ng-container>
+        <ng-container *ngIf="(submitted || duration?.touched) && duration?.hasError('min')">
+          <span id="durationErrorMessage" class="text-danger">Duration must be minimum half hour.</span>
+        </ng-container>
+      </div>
+
+      <div class="course__authors_container">
+        <div class="course__authors_input">
+          <h3>Authors</h3>
+          <div formGroupName="newAuthor">
+            <div class="form__control">
+              <div class="course__flex">
+                <input type="text" id="author" placeholder="Input Author Name" formControlName="author"/>
+                <app-button (click)="addAuthor()" id="createAuthor" [buttonText]="'Create Author'"></app-button>
```

Comment:
As per task requirements, I think we should have a delete option here.
Implementation:
![image](https://github.com/user-attachments/assets/c115cf0b-eb28-49d8-a613-481f85e54e04)

Task:
![image](https://github.com/user-attachments/assets/cec7cd67-9417-4b44-a064-af4b11fde4df)

Reply:
It was intentional: Actually I think the template is wrong or unnecessary as we wont have an author deletion functionality. 
But I added , according to the picture writh no functionality.

Reply:
Hmmm, I see, ok.